gpflow/models/sgpr.py

```diff
@@ -56,11 +57,11 @@ def __init__(
         initialized to `noise_variance`.
         """
         likelihood = likelihoods.Gaussian(noise_variance)
-        X_data, Y_data = data
+        X_data, Y_data = data_input_to_tensor(data)
```

Comment:
```suggestion
        self.data = X_data, Y_data = data_input_to_tensor(data)
```

Reply:
Cannot accept it, as self.data should not exist before `super.__init__`